docs/resources/vpc_network_interface.md

```diff
@@ -0,0 +1,90 @@
+---
+subcategory: "Virtual Private Cloud (VPC)"
+---
+
+# huaweicloud_vpc_network_interface
+
+Manages a VPC network interface resource within HuaweiCloud.
+
+## Example Usage
+
+```hcl
+variable "name" {}
+variable "subnet_id" {}
+
+resource "huaweicloud_vpc_network_interface" "test" {
+  name      = var.name
+  subnet_id = var.subnet_id
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used.
+  Changing this creates a new resource.
+
+* `subnet_id` - (Required, String, ForceNew) Specifies the ID of the network to which the network interface belongs.
+  Changing this creates a new resource.
+
+* `allowed_address_pairs_ips` - (Optional, List) Specifies whether disable the source/destination check.
+  The IP address is "1.1.1.1/0", indicating that the source/destination address check switch is turned off.
+
+* `extra_dhcp_opts` - (Optional, List) Specifies the extended option (extended attribute) of DHCP.
```

Comment:
as we discussed before, it's better to use a string parameter called `dhcp_lease_time`